slam_toolbox:
  ros__parameters:
    use_sim_time: true
    mode: mapping
    map_frame: map
    odom_frame: odom
    base_frame: base_link
    scan_topic: /scan
    publish_tf: true
    max_laser_range: 18.0
    min_laser_range: 0.18
    resolution: 0.05
    minimum_time_interval: 0.10
    transform_publish_period: 0.02
    map_update_interval: 1.0
    map_start_at_dock: false
    enable_interactive_mode: false
    use_scan_matching: true
    use_scan_barycenter: true
    minimum_travel_distance: 0.05
    minimum_travel_heading: 0.02
    scan_buffer_size: 10
    solver_plugin: solver_plugins::CeresSolver
    ceres_linear_solver: SPARSE_NORMAL_CHOLESKY
    ceres_preconditioner: SCHUR_JACOBI
    ceres_use_nonmonotonic_steps: false
    ceres_max_num_iterations: 30
    ceres_num_threads: 4
    ceres_trust_strategy: LEVENBERG_MARQUARDT
